pkg/method/method.go

```diff
@@ -184,7 +184,17 @@ func buildFileMap(ctx context.Context, fetcher remotes.Fetcher, rootDesc ocispec
 func parseURIFields(msg *apt.Message) (ociRef refdocker.Named, title string, err error) {
 	repoURI := msg.Fields[FieldTargetRepoURI]
 	if repoURI == "" {
-		return ociRef, "", fmt.Errorf("missing field %q", FieldTargetRepoURI)
+		if uri := msg.Fields["URI"]; uri == "" {
+			return ociRef, "", fmt.Errorf("missing field %q", FieldTargetRepoURI)
+		} else {
+			// 1. Strip off the proto
+			// 2. Split at the tag (e.g. ":latest" in "ubuntu:latest")
+			// 3. Combine back proto and pre-tag parts
+			// 4. Trim potential trailing file anmes (e.g. "/Packages") and add remainder
+			// 5. Add trailing slash back (important so the "/" is not prepended to file names)
+			split := strings.SplitN(strings.TrimPrefix(uri, "oci://"), ":", 2)
+			repoURI = "oci://" + split[0] + ":" + split[1][:strings.Index(split[1], "/")] + "/"
```

Comment:
Can we split this as `func(string) (string, error)` and have a unit test?


Also could you sign the commit?

Reply:
Done.